### ISSUE TITLE
[BUG]: Not displaying units correctly when user settings differ from project settings

### ISSUE THREAD

**Issue by jtran** [KittyCAD/modeling-app#4230]
### Describe the bug

Changing units would not update the UI in the bottom-right corner. It showed the toast and sliced (which required mm), but the UI widget continued to display meters. No console error. It's correct in the `project.toml` file, but when opening the project in ZMA, it shows "m". No console errors. Switching to _other_ units like feet works. But _not_ millimeters.

So we're pretty sure it's using millimeters correctly. It's just not displaying it correctly.

### Steps to Reproduce

1. Set user units to meters
2. Set project settings to millimeters

### Expected Behavior

Indicator in the bottom-right reflects the units selected in the bottom-right corner menu.

Workaround was to delete the units from the user settings file. Then it worked. So our hypothesis is that the widget is not correctly using the hierarchy of settings.

### Desktop OS

Linux

### Version

0.26.0

**Comment by franknoirot** [comment]
I believe this is no longer the case but I'll check tomorrow

**Comment by franknoirot** [comment]
Nope, I'm able to reproduce the behavior you reported 4+ months ago. So I'm going to fix it right now.

**Comment by franknoirot** [comment]
So this is definitely because we don't serialize the TOML with serde if a setting has the "default value" and `mm` is that for `defaultUnit`. This is bad, and we probably should have just been serializing all project settings anyway or something? I'm not sure because I do stand by what I've said previously about not writing to disk settings that the user has not signaled any opinion about.

But maybe this is all moot anyway, since we're trying to migrate everyone to use per-file annotations for the default unit setting? What do you all think @KittyCAD/frontend?

**Comment by nadr0** [comment]
The current plan for dealing with units is

- Warning from KCL side (waiting for nrc's work)
- unit of measure button always code mods no matter what
- It displays the values in the file. If no value is in file it shows mm
- Project preference for default unit of measure.
- On all edge cases for file creation we use User Profile unit of measure.
- Do not show User profile preference in the button (related to 3.)
- Make sure any KCL - Sample file is loaded into the system with the correct unit of measure (it is being overwritten)
